client/payment-request/blocks/payment-request-express.js

```diff
@@ -53,9 +54,45 @@ const PaymentRequestExpressComponent = ( {
 		return null;
 	}
 
+	let paymentRequestType = '';
+
+	// Check the availability of the Payment Request API first.
+	paymentRequest.canMakePayment().then( ( result ) => {
+		if ( ! result ) {
+			return;
+		}
+
+		// Set the payment request type.
+		if ( result.applePay ) {
+			paymentRequestType = 'apple_pay';
+		} else if ( result.googlePay ) {
+			paymentRequestType = 'google_pay';
+		}
+	} );
+
+	const onPaymentRequestButtonClick = () => {
+		onButtonClick();
+
+		if ( paymentRequestType === 'google_pay' ) {
```

Comment:
What do you think about grouping those payment request types in an object? By doing this we can map the payment request type and its event. We could have something like this:

```js
const paymentRequestTypeEvents = {
  google_pay: wcpayTracks.events.GOOGLEPAY_BUTTON_CLICK,
  apple_pay: wcpayTracks.events.APPLEPAY_BUTTON_CLICK,
};

if ( paymentRequestTypeEvents.hasOwnProperty( paymentRequestType ) ) {
  const event = paymentRequestTypeEvents[ paymentRequestType ];
  wcpayTracks.recordUserEvent( event, { source: 'checkout' } );
}
```

Reply:
Thanks @eduardoumpierre That's a great suggestion!
Updated in https://github.com/Automattic/woocommerce-payments/pull/7268/commits/f392390c45ba64bcab8dc7ea6d8c3d4c7f45a5d7

There's still a bit of duplicated code between blocks and classic pages, while we can move `trackPaymentRequestButtonClick` to `wcpayTracks` to eliminate that, I think we should limit `wcpayTracks` to high-level Tracks functions. So opting to keep what's in https://github.com/Automattic/woocommerce-payments/pull/7268/commits/f392390c45ba64bcab8dc7ea6d8c3d4c7f45a5d7 as is :)

Reply:
Just to confirm: The source content from the second function on `client/payment-request/index.js` changed from `wcpayPaymentRequestParams.button_context` to `checkout`, was it intended?

Reply:
Good catch. That wasn't intentional. Fixed in https://github.com/Automattic/woocommerce-payments/pull/7268/commits/3277a2dc4b6eef96da4234eead7a2828496df8e0